Pluginize

ale0xb opened the pull request:
This PR does several things:

1. Adapts base code to d3 v4 
2. Modernises the examples in `index.html`
3. Proposes a stub for npm's `package.json`

Theoretically with very few further actions people could use this code by running `npm install d3-cartogram` in their projects. Any doubts, let me know. Thanks!

ale0xb commented:
Bump! 👋

adamflorin commented:
@shawnbot Good memory! @ale0xb Let me know if you have any questions, there are just a half-dozen small changes or so…

ale0xb commented:
Hello! 
I just pushed a couple of minor changes addressing the issues you pointed out @shawnbot. 

Regarding @adamflorin PR I think most of the functionality was already implemented. This library is using a rollup  setup that produces umd builds (and thus makes the code compatible with node and the browser). I didn't have time to test it myself but theoretically it should work just as fine as the official d3 plugins do (ie: d3-scale). Trigger and build and look at the first lines of build/d3-cartogram.js, you'll see what I mean.

Anything else let me know, happy to help!

ale0xb commented:
Hello again @shawnbot, just a quick heads up regarding this PR. Anything else missing?

ale0xb commented:
Lovely @shawnbot, thank you very much!
Don't forget to list it in the [D3 Plugins Wiki](https://github.com/d3/d3/wiki/Plugins)
